parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java

```diff
@@ -65,6 +70,180 @@
   private final GenericFixed fixed = new GenericData.Fixed(
       org.apache.avro.Schema.createFixed("fixedCol", null, null, 4),
       "abcd".getBytes(StandardCharsets.UTF_8));
+  @Test
+  public void testMetricsForTopLevelWithMultipleRG() throws Exception {
+    Schema schema = new Schema(
+        optional(1, "booleanCol", BooleanType.get()),
+        required(2, "intCol", IntegerType.get()),
+        optional(3, "longCol", LongType.get()),
+        required(4, "floatCol", FloatType.get()),
+        optional(5, "doubleCol", DoubleType.get()),
+        optional(6, "decimalCol", DecimalType.of(10, 2)),
+        optional(7, "dateCol", DateType.get()),
+        required(8, "timeCol", TimeType.get()),
+        required(9, "timestampCol", TimestampType.withoutZone()),
+        optional(10, "uuidCol", UUIDType.get()),
+        required(11, "fixedCol", FixedType.ofLength(4)),
+        required(12, "binaryCol", BinaryType.get())
+    );
+    int minimumRowGroupRecordCount = 100;
+    List<Record> records = new ArrayList<>(201);
+
+    Record firstRecord = new Record(AvroSchemaUtil.convert(schema.asStruct()));
+    firstRecord.put("booleanCol", true);
+    firstRecord.put("intCol", 3);
+    firstRecord.put("longCol", 5L);
+    firstRecord.put("floatCol", 2.0F);
+    firstRecord.put("doubleCol", 2.0D);
+    firstRecord.put("decimalCol", new BigDecimal("3.50"));
+    firstRecord.put("dateCol", 1500);
+    firstRecord.put("timeCol", 2000L);
+    firstRecord.put("timestampCol", 0L);
+    firstRecord.put("uuidCol", uuid);
+    firstRecord.put("fixedCol", fixed);
+    firstRecord.put("binaryCol", "S".getBytes());
+    records.add(firstRecord);
+
+    Record secondRecord = new Record(AvroSchemaUtil.convert(schema.asStruct()));
+    secondRecord.put("booleanCol", false);
+    secondRecord.put("intCol", Integer.MIN_VALUE);
+    secondRecord.put("longCol", null);
+    secondRecord.put("floatCol", 1.0F);
+    secondRecord.put("doubleCol", null);
+    secondRecord.put("decimalCol", null);
+    secondRecord.put("dateCol", null);
+    secondRecord.put("timeCol", 3000L);
+    secondRecord.put("timestampCol", 1000L);
+    secondRecord.put("uuidCol", null);
+    secondRecord.put("fixedCol", fixed);
+    secondRecord.put("binaryCol", "W".getBytes());
+    records.add(secondRecord);
+
+    Record lastRecord = firstRecord;
+
+    for (int i = 0; i < minimumRowGroupRecordCount * 2; i++) {
```

Comment:
The process for creating the record set seems a bit too complicated to me. Why not base the records on `i` instead of adding 1 to the last record's value? This should also be a helper method.

Reply:
@rdblue thanks for the review !

I have simplified the method to generate the records, will update the PR.
for helper method, as here Im also generating null values, for few columns.
Plus the values we are adding is depending on the schema.
So, not sure should be we have now, or if we have more places where its required then create a helper as well.
Please let me know what do you think.

Reply:
@rdblue thanks for reviewing !
Now the class `TestMetrics` has the test methods now.
So, will add these methods there.